src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Extensions/NonPortable/CustomAttributeInstantiator.cs

```diff
@@ -68,6 +68,7 @@ public static Attribute Instantiate(this CustomAttributeData cad)
             //
             // Found the right constructor. Instantiate the Attribute.
             //
+            Debug.Assert(matchingParameters != null);
             int arity = matchingParameters.Length;
```

Comment:
I think elsewhere in the runtime repo we just do this to avoid unnecessary code (an assert that is just going to crash on the next line is low value):

```suggestion
            int arity = matchingParameters!.Length;
```

Again, this would be better if we could say "NotNullIfMatchingCtorIsNotNull", but that's just the struggle with the C# type system.